engine/src/main/java/io/camunda/zeebe/engine/processing/common/EventTriggerBehavior.java

```diff
@@ -54,19 +56,35 @@ public EventTriggerBehavior(
 
     elementInstanceState = zeebeState.getElementInstanceState();
     eventScopeInstanceState = zeebeState.getEventScopeInstanceState();
+    processState = zeebeState.getProcessState();
 
     variableBehavior =
         new VariableBehavior(zeebeState.getVariableState(), writers.state(), keyGenerator);
   }
 
-  public void unsubscribeFromEvents(final BpmnElementContext context) {
+  private void unsubscribeFromEventSubprocesses(final BpmnElementContext context) {
```

Comment:
> Don't understand what is the subject that is being unsubscribed from the event sub process?
What does it even mean to unsubscribe from an event sub process?

If an interrupting event subprocess is triggered then no other event subprocess, interrupting or non-interrupting, should be triggered afterward. So, we unsubscribe from the event subprocesses to avoid one of these can be triggered (i.e. remove message subscriptions, cancel timers). It's basically for maintenance (i.e. free resources).

> Don't understand the condition isEventSubprocess. Seems to be a misnomer and should be called isStartEventOfEventSubProcess

Yes that is correct. But `isStartEventOfEventSubProcess` sounds to clunky for me. I prefer names that are intuitive, even if they are not 100% correct.

Reply:
Let's discuss this f2f. I am still lost. And I wonder whether I also struggle with understanding in other places of the code base due to seemingly intuitive naming.
The thing I am struggling with in this naming scheme, is what is the subject and what is the object in this sentence.

e.g. 
`catchEventBehavior.unsubscribeFromEvents(context,...)` to me translates into "Unsubscribe this context from any events". So the context is the subject in the sentence, and the events are the object.

But then `unsubscribeFromEventSubprocess(context)` is something I don't understand. What is the subject and what is the object? All I know is that the `context` is later used as a subject when `catchEventBehavior.unsubscribeFromEvents(context,...)` is called. 

So then this method could be called `unsubscribeEventSubProcessFromEvents(context)` or if you want to have it short and intuitive `unsubscribeEventSubProcess(context)` or even shorter `unsubscribe(context)`

But when the name is `unsubscribeFromEventSubprocess` I ask myself: Unsubscribe what??? And then I find it inconsistent to have two methods which if they were longer I guess would be:
```
private void unsubscribe(Events)FromEventSubprocesses(final BpmnElementContext eventSubProcessContext) {
...
    catchEventBehavior.unsubscribe(EventSubProcess)FromEvents(
        eventSubProcessContext,
...
```
Hope this makes it a little bit clearer, why I am confused by the naming.
Same for the predicate method. In my mind the name is not short and intuitive. It is short and imprecise, and therefore confusing.

Reply:
Let's rename the method to `unsubsribeEventSubprocesses` :+1: